rocAL/rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -61,104 +61,127 @@ rocalSequenceRearrange(RocalContext p_context,
     }
     return output;
 }
-/*
-RocalImage  ROCAL_API_CALL
+
+RocalTensor  ROCAL_API_CALL
 rocalRotate(
         RocalContext p_context,
-        RocalImage p_input,
+        RocalTensor p_input,
         bool is_output,
         RocalFloatParam p_angle,
         unsigned dest_width,
-        unsigned dest_height)
-{
+        unsigned dest_height,
+        RocalResizeInterpolationType interpolation_type,
+        RocalTensorLayout rocal_tensor_output_layout,
+        RocalTensorOutputType rocal_tensor_output_datatype) {
 
-    Image* output = nullptr;
+    Tensor* output = nullptr;
     if ((p_context == nullptr) || (p_input == nullptr)) {
         ERR("Invalid ROCAL context or invalid input image")
         return output;
     }
     auto context = static_cast<Context*>(p_context);
-    auto input = static_cast<Image*>(p_input);
+    auto input = static_cast<Tensor*>(p_input);
     auto angle = static_cast<FloatParam*>(p_angle);
-    try
-    {
-        if(dest_width == 0 || dest_height == 0)
-        {
-            dest_width = input->info().width();
-            dest_height = input->info().height_single();
+    try {
+        if(dest_width == 0 || dest_height == 0) {
+            dest_width = input->info().max_shape()[0];
+            dest_height = input->info().max_shape()[1];
         }
-        // For the rotate node, user can create an image with a different width and height
-        ImageInfo output_info = input->info();
-        output_info.width(dest_width);
-        output_info.height(dest_height);
-
-        output = context->master_graph->create_image(output_info, is_output);
+        RocalTensorlayout op_tensor_layout = static_cast<RocalTensorlayout>(rocal_tensor_output_layout);
+        RocalTensorDataType op_tensor_datatype = static_cast<RocalTensorDataType>(rocal_tensor_output_datatype);
 
-        // If the user has provided the output size the dimension of all the images after this node will be fixed and equal to that size
-        if(dest_width != 0 && dest_height != 0)
-            output->reset_image_roi();
+        // For the rotate node, user can create an image with a different width and height
+        TensorInfo output_info = input->info();
+        output_info.set_tensor_layout(op_tensor_layout);
+        output_info.set_data_type(op_tensor_datatype);
+        std::vector<size_t> out_dims = output_info.dims();
+        if(op_tensor_layout == RocalTensorlayout::NHWC) {
```

Comment:
If many functions have assigning dest values wrt tensor layout, can we add it as a common function?

Reply:
Will make the change

Reply:
Done